### PR TITLE
[NextJS]Style guide V1

### PR DESCRIPTION
**Pull request by DavidRojas1612**
## Summary
Initial version of Wolox style guide of Nextjs framework with the principal topics about the tech.

## Known issues / Notes

N/A

## Others ...

N/A

### CODE REVIEW

File: frontend/NextJs/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+## Index
+
+- [Project Bootstrapper](#project-bootstrapper)
+- [Style Guide](#style-guide)
+
+## Project Bootstrapper
+
+At [Wolox](http://wolox.com.ar), we have developed the
+[NextJs Bootstrap](https://github.com/Wolox/next-bootstrap) to create our React
```
Comment (MMarussi) [inline frontend/NextJs/README.md:11]:
```suggestion
[NextJs Bootstrap](https://github.com/Wolox/next-bootstrap) to create our NextJs
```

File: frontend/NextJs/style-guide.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+│
+└───components
+│ └───baseComponents
+│     └───Input
+│     └───Text
+│     └───Button
+│     └───etc
+│
+└───pages
+│   | index.page.tsx // root or hoem page.
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:44]:
```suggestion
│   | index.page.tsx // root or home page.
```

File: frontend/NextJs/style-guide.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+```
+
+## Fetching Data
+
+Next.js allows you to fetch data in different ways, depending on your needs and
+preferences.
+
+- **getStaticProps**: The getStaticProps function is executed on the server side
+  is compiled, and allows you to retrieve data from an external API or any other
+  source and pass it as properties to the component. This function is util when
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:83]: I think that "util" as an adjective does not exist in english (idk if "util" may be used as a noun such as the "utils" directory on some projects though)
```suggestion
  source and pass it as properties to the component. This function is useful when
```

File: frontend/NextJs/style-guide.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  we need to have some pages prebuilt like **blogs**, or **lading
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:84]:
```suggestion
  we need to have some pages prebuilt like **blogs**, or **landing
```

File: frontend/NextJs/style-guide.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+└───utils
+│   index.ts
+```
+
+## Fetching Data
+
+Next.js allows you to fetch data in different ways, depending on your needs and
+preferences.
+
+- **getStaticProps**: The getStaticProps function is executed on the server side
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:81]: How about this? The function description may result a little bit redundant if we name the function after it's "section title"

```suggestion
- **getStaticProps**: This function is executed on the server side
```

Comment (MMarussi) [reply frontend/NextJs/style-guide.md:81]: The same applies to other functions

File: frontend/NextJs/style-guide.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+    const data = await res.json()
+
+    // Pass the data as props to the component
+    return {props: {data}}
+  }
+  ```
+
+- **getServerSideProps**: The getServerSideProps function, which is executed on
+  the server side every time a page is accessed, to fetch data and pass it to
+  the component as properties. yop can also get the query parameters in the URL
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:107]:
```suggestion
  the component as properties. you can also get the query parameters in the URL
```

File: frontend/NextJs/style-guide.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+  ```js
+  function MyComponent(props) {
+    return (
+      // Use the retrieved data as component props
+      <div>{props.data}</div>
+    )
+  }
+
+  export async function getServerSideProps(context) {
+    // Retrieve data from theexternal API
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:119]:
```suggestion
    // Retrieve data from the external API
```

File: frontend/NextJs/style-guide.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+data to the console.
+
+You can read more examples
+[here](https://github.com/vercel/next.js/tree/canary/examples/api-routes-rest)
+You can read the docs [here](https://nextjs.org/docs/api-routes/introduction)
+
+## SEO
+
+In Next.js you can improve the SEO of the websites using some
+[pre-rendering modes](#fetching-data) this modes will we could to get a document
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:182]: I think that the redaction here is a little bit messy, don't know if I'm getting the right idea, how about?
```suggestion
[pre-rendering modes](#fetching-data) these modes will generate the application
```

File: frontend/NextJs/style-guide.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+HTML optimized for the web crawler and indexers.
+
+- **Prerendering modes**:
+
+  - Static Generation: HTML is only generating during build time.
+
+  - Server-side Rendering: HTML is generating on each reques.<br /><br />
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:189]:
```suggestion
  - Static Generation: HTML is only generated during build time.

  - Server-side Rendering: HTML is generated on each request.<br /><br />
```

File: frontend/NextJs/style-guide.md [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+    )
+  }
+
+  export default IndexPage
+  ```
+
+  Head Component makes sure all of your pages include important details like
+  title, description etc. that need to get rendered into your page. Next.js
+  recognises this metadata and lift it to the right location in your HTML
+  document when the page is being rendered. It helps in improving SEO.
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:216]: How about this
```suggestion
  Head Component helps us to improve SEO by making sure all of your pages include important details like
  title, description, etc. that need to get rendered into your page. Next.js
  recognises this metadata and lift it to the right location in your HTML
  document when the page is being rendered.
```

File: frontend/NextJs/style-guide.md [hunk 11]
```diff
@@ -0,0 +1,219 @@
+# Next.js Style Guide
+
+## Table of Contents
+
+1. [Basic Rules](#basic-rules)
+1. [Folder Structure](#folder-structure)
+1. [Fetching Data](#fetching-data)
+1. [Api Routes](#naming)
+1. [SEO](#mixins)
```
Comment (MMarussi) [inline frontend/NextJs/style-guide.md:9]: These links should be updated

File: frontend/NextJs/style-guide.md (hunk 8 above)
Comment (ivansevillaa) [reply frontend/NextJs/style-guide.md:182]: Hi @MMarussi! what about `these modes will let you get an optimized HTML document`?

Comment (MMarussi) [reply frontend/NextJs/style-guide.md]: Perfect!

**Review by MMarussi — APPROVED** [review]

**Pull request merged**